Mathlib/Logic/Basic.lean

```diff
@@ -1031,7 +1031,11 @@ theorem some_spec₂ {α : Sort*} {p : α → Prop} {h : ∃ a, p a} (q : α →
     (hpq : ∀ a, p a → q a) : q (choose h) := hpq _ <| choose_spec _
 #align classical.some_spec2 Classical.some_spec₂
 
-/-- A version of `Classical.indefiniteDescription` which is definitionally equal to a pair -/
+/-- A version of `Classical.indefiniteDescription` which is definitionally equal to a pair.
+
+In Lean 4, this definition is defeq to `Classical.indefiniteDescription`,
+so it is deprecated. -/
```

Comment:
`example : @subtype_of_exists = @Classical.indefiniteDescription := rfl` fails, so I'm not sure I understand your claim here.